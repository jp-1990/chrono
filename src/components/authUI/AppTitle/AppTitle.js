import React from "react";
import { View, Text, StyleSheet } from "react-native";

import base from "../../../styles/base";

const AppTitle = ({ title1, title2, subtitle }) => {
  return (
    <View style={styles.titleBox}>
      <Text style={styles.title}>{title1}</Text>
      <Text style={styles.title}>{title2}</Text>
      <Text style={styles.subtitle}>{subtitle}</Text>
    </View>
  );
};

export default AppTitle;

const styles = StyleSheet.create({
  titleBox: { padding: 54 },
  title: {
    fontFamily: "lato-light",
    textTransform: "uppercase",
    fontSize: 42,
    color: base.colors.headingPrimary,
  },
  subtitle: {
    fontFamily: "lato-light",
    color: base.colors.textSecondary,
    fontSize: 18,
  },
});
