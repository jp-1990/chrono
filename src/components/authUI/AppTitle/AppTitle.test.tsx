import React from "react";
import TestRenderer from "react-test-renderer";
import { cleanup, render } from "@testing-library/react-native";

import AppTitle from "./AppTitle";

describe("<AppTitle />", () => {
  afterEach(cleanup);

  const testProps = {
    title1: "title1",
    title2: "title2",
    subtitle: "subtitle",
  };
  const testRender = TestRenderer.create(<AppTitle {...testProps} />);

  it("renders correctly", () => {
    expect(testRender).toMatchSnapshot();
  });

  it("contains title1 prop", () => {
    const { getByText } = render(<AppTitle {...testProps} />);
    expect(getByText("title1")).toBeTruthy();
  });

  it("contains title2 prop", () => {
    const { getByText } = render(<AppTitle {...testProps} />);
    expect(getByText("title2")).toBeTruthy();
  });

  it("contains subtitle prop", () => {
    const { getByText } = render(<AppTitle {...testProps} />);
    expect(getByText("subtitle")).toBeTruthy();
  });
});
