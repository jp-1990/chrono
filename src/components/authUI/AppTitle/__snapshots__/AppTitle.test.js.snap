// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTitle /> renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 54,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#705E57",
        "fontFamily": "lato-light",
        "fontSize": 42,
        "textTransform": "uppercase",
      }
    }
  >
    title1
  </Text>
  <Text
    style={
      Object {
        "color": "#705E57",
        "fontFamily": "lato-light",
        "fontSize": 42,
        "textTransform": "uppercase",
      }
    }
  >
    title2
  </Text>
  <Text
    style={
      Object {
        "color": "#818CA0",
        "fontFamily": "lato-light",
        "fontSize": 18,
      }
    }
  >
    subtitle
  </Text>
</View>
`;
